src/RemoteHandler/ServerMock.js

```diff
@@ -8,22 +8,132 @@ class ServerMock {
         this.productFamilies = ServerMock.getProductFamily();
         this.eventTypes = ServerMock.getEventTypes();
         this.events = ServerMock.getEvents();
-        
+        this.productConfigurations = ServerMock.getProductConfigurations();
+
         //function binding
         this.deleteEventType = this.deleteEventType.bind(this);
     }
-    
+
     open(requestMethod, api) {
         this.api = api;
     }
-    
+
     setRequestHeader(key, value) {
         this.headers.push({
             key: key,
             value: value
         });
     }
-    
+
+    static getProductConfigurations() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This function has 108 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS138)